plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java

```diff
@@ -314,17 +314,39 @@ void deleteBlobs(Collection<String> blobNames) throws IOException {
             return;
         }
         final List<BlobId> blobIdsToDelete = blobNames.stream().map(blob -> BlobId.of(bucketName, blob)).collect(Collectors.toList());
-        final List<Boolean> deletedStatuses = SocketAccess.doPrivilegedIOException(() -> client().delete(blobIdsToDelete));
+        final List<Boolean> deletedStatuses = SocketAccess.doPrivilegedIOException(() -> {
+            final StorageBatch batch = client().batch();
+            final List<Boolean> results = Lists.newArrayList();
+            for (BlobId blob : blobIdsToDelete) {
+                batch.delete(blob).notify(
+                    new BatchResult.Callback<>() {
+                        @Override
+                        public void success(Boolean result) {
+                            results.add(result);
+                        }
+
+                        @Override
+                        public void error(StorageException exception) {
+                            if (exception.getCode() == 404) {
```

Comment:
let's use the constant here: `HttpURLConnection.HTTP_NOT_FOUND`

Reply:
sure :)